tests/test_timeout.py

```diff
@@ -344,3 +345,51 @@ async def test_deprecated_with() -> None:
     with pytest.warns(DeprecationWarning):
         with timeout(1):
             await asyncio.sleep(0)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 7), reason="Not supported in 3.6")
+@pytest.mark.asyncio
+async def test_race_condition_cancel_before() -> None:
+    """Test race condition when cancelling before timeout.
+
+    If cancel happens immediately before the timeout, then
+    the timeout may overrule the cancellation, making it
+    impossible to cancel some tasks.
+    """
+
+    async def test_task(deadline: float, loop: asyncio.AbstractEventLoop) -> None:
+        with Timeout(deadline, loop):
```

Comment:
This has broken the tests. `timeout()` uses a delay, whereas the internal `Timeout` uses a deadline. We need to use deadline to ensure the timing is exact, otherwise the race condition isn't reproducible.

I'll revert and add a comment in.